pkg/scheduling/v1/queuer.go

```diff
@@ -54,24 +68,43 @@ func newQueuer(conf *sharedConfig, tenantId pgtype.UUID, queueName string, s *Sc
 		defaultLimit = conf.singleQueueLimit
 	}
 
+	cacheSize := 1000
+	if conf.singleQueueLimit > 1000 {
+		cacheSize = conf.singleQueueLimit
+	}
```

Comment:
The `singleQueueLimit` has nothing to do with the actual number of queues, it's the amount of rows we read out of each queue, so it doesn't make sense to have these values be correlated